### PR TITLE
src/xwayland-unmanaged.c: Prevent racing configure event to reposition

### LINKED ISSUE #469: labwc crashes when starting wine game (unmanaged surface?)
started recently-ish? used to work fine iirc. 
game in question is league of legends - starting the launcher works fine but when opening the client where usually a splash screen appears labwc crashes.

```
../subprojects/wlroots/types/scene/wlr_scene.c:605:10: runtime error: member access within null pointer of type 'struct wlr_scene_node'
AddressSanitizer:DEADLYSIGNAL
=================================================================
==18425==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000024 (pc 0x0000005b2188 bp 0x7fff5f21c5d0 sp 0x7fff5f21c5c0 T0)
==18425==The signal is caused by a READ memory access.
==18425==Hint: address points to the zero page.
    #0 0x5b2188 in wlr_scene_node_set_position ../subprojects/wlroots/types/scene/wlr_scene.c:605
    #1 0x476522 in unmanaged_handle_request_configure ../src/xwayland-unmanaged.c:13
    #2 0x6a5d90 in wlr_signal_emit_safe ../subprojects/wlroots/util/signal.c:29
    #3 0x6c1816 in xwm_handle_configure_request ../subprojects/wlroots/xwayland/xwm.c:980
    #4 0x6c9b23 in x11_event_handler ../subprojects/wlroots/xwayland/xwm.c:1610
    #5 0x7fc199853dd1 in wl_event_loop_dispatch (/lib64/libwayland-server.so.0+0xbdd1)
    #6 0x7fc199854514 in wl_display_run (/lib64/libwayland-server.so.0+0xc514)
    #7 0x43a8da in main ../src/main.c:108
    #8 0x7fc19876c5af in __libc_start_call_main (/lib64/libc.so.6+0x405af)
    #9 0x7fc19876c678 in __libc_start_main_alias_1 (/lib64/libc.so.6+0x40678)
    #10 0x416674 in _start (/home/florian/src/labwc/debug/build/labwc+0x416674)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV ../subprojects/wlroots/types/scene/wlr_scene.c:605 in wlr_scene_node_set_position
==18425==ABORTING
(EE) failed to read Wayland events: Connection reset by peer
```

### PR DESCRIPTION
**Pull request by Consolatis**
Fixes #469

@Flrian testing welcome.

@johanmalm @jlindgren90
This is not an area of the codebase I am inherently familiar with so extra caution should be taken for the review.

### CODE REVIEW

File: src/xwayland-unmanaged.c [hunk 1]
```diff
@@ -79,6 +82,11 @@ unmanaged_handle_unmap(struct wl_listener *listener, void *data)
 			return;
 		}
 	}
+	/*
+	 * Mark the node as gone so a racing configure event
+	 * won't try to reposition the node while unmapped.
+	 */
+	unmanaged->node = NULL;
```
Comment (jlindgren90) [inline src/xwayland-unmanaged.c:89]: This should probably be moved up before any of the `return;` paths.  Perhaps right near the two `wl_list_remove()` calls.

**Comment by jlindgren90** [comment]
@Consolatis Thanks for working on this. One comment above, otherwise +1 from me.

File: src/xwayland-unmanaged.c (hunk 1 above)
Comment (Joshua-Ashton) [reply src/xwayland-unmanaged.c:89]: This then LGTM

Comment (Consolatis) [reply src/xwayland-unmanaged.c]: > This should probably be moved up before any of the return; paths. Perhaps right near the two wl_list_remove() calls.

Should really have seen that, updated. Thanks for the review!

**Comment by johanmalm** [comment]
Thanks all.

**Pull request merged**